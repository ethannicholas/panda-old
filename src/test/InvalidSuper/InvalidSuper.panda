class Test {
    constructor() {
        super()
    }

    function one():Bit {
        return super = self
    }

    function two(t:Test):Bit {
        return super.hash-?>Int
    }

    function three(t:Test):Bit {
        return true
    }

    function four():Bit {
        return two(super)
    }

    @class
    method main() {
        Console.writeLine(super)
        Console.writeLine(super.hash)
    }
}