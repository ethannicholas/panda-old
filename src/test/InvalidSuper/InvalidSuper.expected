error: InvalidSuper.panda:3:9: 'super' is not a method
error: InvalidSuper.panda:7:16: 'super' is not valid in this context
error: InvalidSuper.panda:19:20: 'super' is not valid in this context
error: InvalidSuper.panda:24:27: 'super' is not valid in this context
error: InvalidSuper.panda:25:32: unknown identifier
5 errors
